erpnext/selling/page/point_of_sale/point_of_sale.py

```diff
@@ -119,6 +109,23 @@ def get_items(start, page_length, price_list, item_group, search_value="", pos_p
 
 	return res
 
+@frappe.whitelist()
+def search_serial_or_batch_or_barcode_number(search_value):
+	# search serial no
+	serial_no_data = frappe.db.get_value('Serial No', search_value, ['name as serial_no', 'item_code'], as_dict=True)
+	if serial_no_data:
```

Comment:
Any reason why the searching sequence is serial number , then batch, finally barcode, not the other way around?
There is also barcode field in Item table, it is already obsolete and replaced by the Item barcode child table?